course/templates/course/index.html

```diff
@@ -12,85 +12,144 @@ <h1>{{ welcome_text|safe }}</h1>
+                    </div>
+                </a>
                 <div class="caption">
-                    <h3>{{ advert.title }}</h3>
-                    <p>{{ advert.text|safe }}</p>
+                    <h3>
+                        <a href="{{ instance_url }}">
+                            {{ instance.course.name|parse_localization }}<br />
+                        </a>
+                    </h3>
+                    <p>
+                        {{ instance.course.code }}
+                        {% if not instance.visible_to_students %}
+                        <span class="label label-danger">{% trans "Hidden from students" %}</span>
+                        {% endif %}
+                        <br />
+                        <!--{{ instance.instance_name }}<br />-->
+                        {{ instance.starting_time|date:"SHORT_DATE_FORMAT" }} &ndash;
+                        {{ instance.ending_time|date:"SHORT_DATE_FORMAT" }}
+                        <br />
+                        {% if instance.enrollment_audience == 1 or instance.enrollment_audience == 3 %}
+                        <span class="label label-success">{{ internal_user_label|safe }}</span>
+                        {% endif %}
+                        {% if instance.enrollment_audience == 2 or instance.enrollment_audience == 3 %}
+                        <span class="label label-info">{{ external_user_label|safe }}</span>
+                        {% endif %}
+                    </p>
                     <div class="buttons caption">
-                        <a class="btn btn-default btn-block" role="button" href="{{ advert.href }}" target="_blank">
-                            <span class="glyphicon glyphicon-new-window" aria-hidden="true"></span>
-                            {% trans "Read more" %}
+                        <a class="btn btn-primary btn-block" role="button" href="{{ instance_url }}">
+                            {% trans "View course" %}
                         </a>
                     </div>
                 </div>
             </div>
         </div>
-    {% endif %}
+        {% endwith %}
+        {% endfor %}
+    </div>
+</div>
+{% endif %}
 
-    {% for instance in instances %}
-    {% with instance_url=instance|url %}
-    <div class="col-sm-6 col-md-4 col-lg-3">
-        <div class="thumbnail course-thumbnail">
-            <a href="{{ instance_url }}">
-                <div class="image" aria-hidden="true">
-                    {% if instance.image %}
-                    <img src="{{ instance.image.url }}" alt="{{ instance.course.name|parse_localization }}" />
-                    {% else %}
-                    <h1 class="glyphicon glyphicon-book"></h1>
-                    {% endif %}
+<div class="frontpage-panel">
+    {% if all_instances or site_advert %}
+    <h2 class="frontpage panel-heading">{% trans "All courses" %}</h2>
+    <div class="row course-thumbnails">
+        {% site_advert as advert %}
```

Comment:
I thought about this. My idea was that since not all are logged in straightaway, this way they could get to see the advert as well. 

If the advert would be in the my courses -section, when a not logged-in person comes to the front page, they would see only an otherwise empty "my courses"-section with the advert, and then below that the section for all courses. 

I also thought about separating the advert as an own separate instance on the right-hand side, but that seemed like out of the scope of this issue. And something that could maybe be done when the filtering and search are implemented.

Reply:
Yeah. I didn't consider it could be bit complicated to show the advert in both cases, which would require to add it to different div in each case. I think your solution is fine.